appcontainer/BannerSourceModel.ts

```diff
@@ -22,17 +22,31 @@ export class BannerSourceModel extends HoistModel {
     @observable.ref
     bannerModels: BannerModel[] = [];
 
-    MAX_BANNERS = 4;
-
     constructor() {
         super();
         makeObservable(this);
     }
 
     @action
     show(config: BannerSpec): BannerModel {
-        const ret = new BannerModel(config);
-        this.addModel(ret);
+        let {bannerModels} = this,
+            {category} = config,
+            existing = category ? bannerModels.find(it => it.category == category) : null,
+            sortOrder;
+
+        // Removes banner from new banner's category if one exists.
+        // Otherwise, creates the banner and adds it as the bottom banner.
+        if (existing) {
+            bannerModels = without(bannerModels, existing);
+            XH.safeDestroy(existing);
+            sortOrder = config.sortOrder ?? existing.sortOrder;
+        } else {
+            const maxSortOrder = maxBy(bannerModels, 'sortOrder')?.sortOrder ?? 0;
+            sortOrder = config.sortOrder ?? maxSortOrder + 1;
+        }
+        const ret = new BannerModel({...config, sortOrder});
+        this.bannerModels = sortBy([...bannerModels, ret], 'sortOrder');
+
```

Comment:
Changed the show() method to account for updating a message in the same category and keeping the banner order stable.